Add customization options section (#127)
bump-sh/docs: src/_help/customization-options/custom-domains.md

---
title: Custom domains
---

- TOC
{:toc}

> Your API documentation is hosted on Bump.sh under our URL. To extend your brand experience to your visitors and API consumers, you can use a custom domain that replaces our URL. This change is based on a CNAME record.

Reviewer: ```suggestion
Your API documentation is hosted on Bump.sh under our domain name. By default, accessing your documentation will be possible via a URL such as `https://bump.sh/<user_or_organization_name>/doc/<doc_slug>`.

To extend your brand experience to your visitors and API consumers, you can use a custom domain that replaces our URL. This change is based on a CNAME record.
```

Author: Partially added

A CNAME, or Canonical Name, is a Domain Name System (DNS) entry. It serves as an alias, and maps a domain name to another. In our use case, it lets you set a domain name for your users to access your API docs, that they will use and see instead of the Bump.sh domain name.

## Setting a CNAME record

First, you'll need to create a CNAME record pointing to `custom.bump.sh`, at your domain name provider.

For instance, if you want to host your documentation under the `developers.example.com` domain, you will create the following record:

```undefined
developers.example.com. 3600 IN CNAME custom.bump.sh.
```

> The configuration of the CNAME record can vary from one hosting provider to another. Feel free to contact your hosting provider if you have any questions at this stage.
{: .info}

Once this is done, you can set your custom domain in Bump.sh.

## Setting a custom domain in Bump.sh

> From the settings of your documentation, check the "Custom domain" box. You will then be asked to enter the URL of your custom domain, which should have been configured on your end in the previous step. Confirm by selecting "Update General Settings".

Reviewer: ```suggestion
From the settings of your documentation, check the "Custom domain" box. You will then be asked to enter the URL of your custom domain, which has been configured on your end in the previous step. Confirm by selecting "Update General Settings".
```
I wouldn't use the conditional form here, and keep it more neutral.

Author: I'd rather use the conditional form, as this it not under our control.

![](/images/help/custom-domains.png)

> SSL certificates are automatically issued.
{: .info}

> It may take some time for your custom domain update to propagate. This delay is normal and is not dependent on Bump.sh or your hosting provider.

> During this period, visitors returning to access your documentation may have difficulty accessing it. It is recommended to try again later or clear your cache before making another attempt.

Reviewer: The cache refresh recommendation is relevant only to users who have previously visited the documentation, before the custom domain was set. It won't apply to all "brand new visitors". Not sure how/if we should clarify this here.

Author: I hesitated a lot on that one but I felt that adding this could generate confusion

---
title: Custom domains
---

- TOC
{:toc}

Your API documentation is hosted on Bump.sh under our domain name. By default, accessing your documentation will be possible via a URL such as `https://bump.sh/<user_name>/doc/<doc_slug>`.

To extend your brand experience to your visitors and API consumers, you can use a custom domain that replaces our URL. This change is based on a CNAME record.

> A CNAME, or Canonical Name, is a Domain Name System (DNS) entry. It serves as an alias, and maps a domain name to another. In our use case, it lets you set a domain name for your users to access your API documentation, that they will use and see instead of the Bump.sh domain name.

Author: Initially, I added that part for non-technical users but I agree.

## Setting a CNAME record

First, you'll need to create a CNAME record pointing to `custom.bump.sh`, at your domain name provider.

For instance, if you want to host your documentation under the `developers.example.com` domain, you will create the following record:

```undefined
developers.example.com. 3600 IN CNAME custom.bump.sh.
```

> The configuration of the CNAME record can vary from one hosting provider to another. Feel free to contact your hosting provider if you have any questions at this stage.
{: .info}

Once this is done, you can set your custom domain in Bump.sh.

## Setting a custom domain in Bump.sh

> From the settings of your documentation, check the "Custom domain" box. You will then be asked to enter the URL of your custom domain, which should have been configured on your end in the previous step. Confirm by selecting "Update General Settings".

Author: I'm really sorry, I didn't get that comment. What do you mean by "attribute"?

> From the settings of your documentation, check the "Custom domain" box. You will then be asked to enter the URL of your custom domain, which should have been configured on your end in the previous step. Confirm by selecting "Update General Settings".

Reviewer 2: Let's forget this comment. It was about using codeblocks vs simple quotes, but we want get back to it later (if needed).
